### PR TITLE
CP-2183-Enhancement of flutter SDK by adding required public methods.

### PR DESCRIPTION
**Pull request by Azharhussain20**
CP-2183-Flutter SDK Support.
File changes.
- `CleverPushPlugin.m`
- `cleverpush_flutter.dart`
- `main.dart`

 Refactor `main.dart` , `cleverpush_flutter.dart` & `CleverPushPlugin.m`  by adding four more methods.
 - `Get Notification`
 - `Get Subscription Topics`
 - `Set Subscription Topics`
 - `Get Available Topics`

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -38,14 +40,14 @@ class _MyAppState extends State<MyApp> {
...
-        _debugLabelString = "Subscribed: " + subscriptionId;
+        _debugLabelString = "Subscribed: " + subscriptionId!;
       });
 
       print("Subscribed: ${subscriptionId}");
     });
 
     // CleverPush Channel ID
-    await CleverPush.shared.init("hxWyS7jPk4DrnSk5K", true);
+    await CleverPush.shared.init("64ipj2EG2gGNGkEr7", true);
```
Comment (gbhrdt) [inline example/lib/main.dart:50]: Please revert :-)

File: example/lib/main.dart [hunk 2]
```diff
@@ -69,6 +71,46 @@ class _MyAppState extends State<MyApp> {
...
+        topicIdsString = topicIdsString + "," + topicIds[i];
+      }
+    }
+    this.setState(() {
+      _debugLabelString = topicIdsString;
+    });
+  }
+
+  void _setSubscriptionTopics() {
+    List<String> topics = ['hello', 'world'];
```
Comment (gbhrdt) [inline example/lib/main.dart:100]: Topics need to be given via parameter

File: ios/Classes/CleverPushPlugin.m [hunk 3]
```diff
@@ -20,15 +20,15 @@ + (instancetype)sharedInstance {
 
 + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
     /*
-    [CleverPush initWithLaunchOptions:nil channelId:nil handleNotificationOpened:^(CPNotificationOpenedResult *result) {
-        CleverPushPlugin.sharedInstance.coldStartOpenResult = result;
-    }];
-    */
-
+     [CleverPush initWithLaunchOptions:nil channelId:nil handleNotificationOpened:^(CPNotificationOpenedResult *result) {
```
Comment (gbhrdt) [inline ios/Classes/CleverPushPlugin.m:23]: Remove comment

File: example/lib/main.dart [hunk 4]
```diff
@@ -1,6 +1,6 @@
+import 'dart:ffi';
```
Comment (gbhrdt) [inline example/lib/main.dart:1]: Sort imports (both dart imports under each other)

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: example/lib/main.dart (hunk 1 above)
Comment (Azharhussain20) [reply example/lib/main.dart:50]: Ok

File: example/lib/main.dart (hunk 2 above)
Comment (Azharhussain20) [reply example/lib/main.dart:100]: Only methods name can be accessible from the `CleverPushButton` that is why I have made a methods without parameter.

Comment (gbhrdt) [reply example/lib/main.dart:100]: Sorry, you're right, I missed that it's the example class.

**Pull request merged**